tests/helpers/reporter/reporter.go

```diff
@@ -0,0 +1,211 @@
+
+package main
+
+import (
+	"github.com/01org/cc-oci-runtime/tests/helpers/reporter/metric"
+)
+
+var LogPath string
+
+func writeJSONEnvironment() {
+	const JSONFile = "environment"
+	environment := metric.GetEnvironment()
+	environment.WriteJSONEnvironment(JSONFile)
+}
+
+func writeTestResult() {
+
+	//Write JSON for Docker Test Results
+
+	dockerTestRegex := metric.TestRegex{
+		LogFile: LogPath + "docker-integration-tests.html",
+		//		JSONFile:  "docker_test_result.json",
+		SkipField: "# skip",
+		FailField: ">not ok ",
+		PassField: ">ok ",
+	}
+
+	dockerTestResult := metric.Test{
+		Name:        "docker test result",
+		Description: "docker test result for cc-oci-runtime",
+		Type:        "integration tests",
+	}
+
+	dockerTest :=
+		metric.GetTestResult(dockerTestRegex, dockerTestResult)
+	dockerTest.WriteJSONTestResult()
+
+	//Write JSON for Unit Test Results
+
+	unitTestRegex := metric.TestRegex{
+		LogFile: LogPath + "unit_tests.log",
+		//		JSONFile:  "unit_test_result.json",
+		SkipField: "^SKIP",
+		FailField: "^FAIL",
+		PassField: "^PASS",
+	}
+
+	unitTestResult := metric.Test{
+		Name:        "unit test result",
+		Description: "unit tests for cc-oci-runtime",
+		Type:        "unit tests",
+	}
+
+	unitTest := metric.GetTestResult(unitTestRegex, unitTestResult)
+	unitTest.WriteJSONTestResult()
+
+	//Write JSON for Functional Test Results
+	functionalTestsRegex := metric.TestRegex{
+		LogFile: LogPath + "functional_tests.log",
+		//		JSONFile:  "functional_test_result.json",
+		SkipField: "# skip",
+		FailField: "^not ok",
+		PassField: "^ok ",
+	}
+
+	testResult := metric.Test{
+		Name:        "functional test result",
+		Description: "functional tests result for cc-oci-runtime",
+		Type:        "functional tests",
+	}
+
+	functionalTest :=
+		metric.GetTestResult(functionalTestsRegex, testResult)
+	functionalTest.WriteJSONTestResult()
+
+	//Write JSON for Valgrind test results
+	valgrindRegex := metric.TestRegex{
+		LogFile: LogPath + "valgrind_tests.log",
+		//		JSONFile:  "valgrind_test_result.json",
+		SkipField: "^SKIP:",
+		FailField: "^FAIL:",
+		PassField: "^PASS:",
+	}
+
+	valgrindResult := metric.Test{
+		Name:        "valgrind test result",
+		Description: "valgrind for cc-oci-runtime",
+		Type:        "valgrind tests",
+	}
+
+	valgrind := metric.GetTestResult(valgrindRegex, valgrindResult)
+	valgrind.WriteJSONTestResult()
+
+	//Write JSON for code Coverage
+	codeCoverageRegex := metric.CodeRegex{
+		LogFile: LogPath + "test_summary.log",
+		//	JsonFile:
+		LineCoverageField:     "Lines Coverage: ",
+		FunctionCoverageField: "Functions Coverage: ",
+	}
+
+	codeCoverageResult := metric.Code{
+		Name: "code_coverage",
+		Type: "Code coverage",
+	}
+
+	codeCoverage :=
+		metric.GetCodeCoverage(codeCoverageRegex, codeCoverageResult)
+	codeCoverage.WriteJSONCodeCoverage()
+}
+
+func writeReturnCode() {
+
+	//Return codes are stored in the test_summary.log
+
+	var LogFile = LogPath + "test_summary.log"
+	const (
```

Comment:
This set of `const`s and the calls to `GetJSONReturnCode()` and `WriteJSONReturnCode()` below would lend themselves nicely to a table-based approach I think: define a struct for the `Name` and `RE`, create an array of the structs initialising the entries to the values below, then iterating through the array calling the two functions on each entry.